cmake/onnxruntime.cmake

```diff
@@ -39,6 +39,13 @@ else()
   set(ONNXRUNTIME_SO_LINK_FLAG "-DEF:${SYMBOL_FILE}")
 endif()
 
+if (NOT WIN32)
+  set(CMAKE_SKIP_BUILD_RPATH  FALSE)
```

Comment:
Why ???

Reply:
This is from snippet below. ML.NET uses these options for both Linux and Mac.

https://github.com/dotnet/machinelearning/blob/cde7038b3132801ee212b3757945a3697a1f485c/src/Native/MklProxyNative/CMakeLists.txt#L11-L14

Reply:
I don't think these settings are good for the C package. It's a magic that we can't fully understand and I still doesn't know why it is needed. Such code will make this project hard to maintain, and it may result unexpected behavior for the C API users.

Reply:
Reduced the number of flags to the minimum needed.

Reply:
One big difference between Windows and Linux is: Linux has suid programs. I would think this is why Linux takes RPATH so seriously.
If:
1. There is an suid program, who linked to a shared library with rpath setting
2. And a low privilege user can write to that rpath dir
Then the low privilege user can gain root access by putting a fake libc.so into that rpath dir.

Therefore we must carefully clear out the RPATH settings in libonnxruntime.so before making the package.  "$ORIGIN" is acceptable. The others are not.